src/main/java/uk/ac/cam/cl/dtg/isaac/api/EventsFacade.java

```diff
@@ -896,16 +952,22 @@ public final Response recordEventAttendance(@Context final HttpServletRequest re
                                                 @PathParam("user_id") final Long userId,
                                                 @QueryParam("attended") final Boolean attended) {
         try {
-            if (!isUserAnAdminOrEventManager(userManager, request)) {
+            if (!isUserAbleToManageEvents(userManager, request)) {
                 return new SegueErrorResponse(Status.FORBIDDEN, "You must be a staff user to access this endpoint.")
                         .toResponse();
             }
 
+            RegisteredUserDTO currentUser = this.userManager.getCurrentRegisteredUser(request);
             RegisteredUserDTO userOfInterest = this.userManager.getUserDTOById(userId);
             IsaacEventPageDTO event = this.getEventDTOById(request, eventId);
 
+            if (currentUser.getRole() == Role.EVENT_LEADER && !userAssociationManager.hasPermission(currentUser, userOfInterest)) {
```

Comment:
Do we not need to check if the event leader is trying to manage the attendance for an event they control? I wonder if it is possible for a student to have a valid association with a event leader and for the event leader to change the attendance status of a student on an event they have nothing to do with? Perhaps this should be a standard check on the event leader permissions check? i.e is the event anything to do with the leader?